src/cmdline/mod.rs

```diff
@@ -209,6 +220,114 @@ impl Cmdline {
     pub fn as_str(&self) -> &str {
         self.line.as_str()
     }
+
+    /// Adds a serial console device to the kernel command line.
+    ///
+    /// Multiple devices can be specified, with multiple `console=` options. This function must be
+    /// called once per device.
+    /// No checks are performed on the device name. The caller must ensure a valid one.
+    /// Multiple options can be specified per device.
+    /// For more details see the [documentation] (section `console=`).
+    ///
+    /// # Arguments
+    ///
+    /// * `device` - guest device where the console output will go.
+    /// * `options` - optional additional configurations for the device (e.g. baud rate).
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use linux_loader::cmdline::*;
+    /// # use std::ffi::CString;
+    /// let mut cl = Cmdline::new(50);
+    /// cl.add_serial_console("ttyS0", vec!["9600"]).unwrap();
+    /// let cl_cstring = CString::new(cl).unwrap();
+    /// assert_eq!(cl_cstring.to_str().unwrap(), "console=ttyS0,9600");
+    /// ```
+    ///
+    /// [documentation]: https://www.kernel.org/doc/html/v4.16/admin-guide/kernel-parameters.html
+    pub fn add_serial_console<T: AsRef<str>>(&mut self, device: T, options: Vec<T>) -> Result<()> {
```

Comment:
This should be more generally `add_console` as we can use it to specify `hvc0` which is the virtio-console.